### PR TITLE
added orderby

### PR DESCRIPTION
**Pull request by alfredkam**

### CODE REVIEW

File: lib/Database.js [hunk 1]
```diff
@@ -149,6 +152,20 @@ var Database = module.exports = Class.extend({
...
   /**
+   * Create the order by query segment
+   */
+  getOrderBy: function (orders) {
+    if (typeof orders == 'string' || orders instanceof String) return ' ORDER BY ' + orders;
+    if (!orders || (orders && orders.length === 0)) return '';
+    var query = ' ORDER BY';
+    for (var i in orders) {
+      var order = orders[i];
+      query += ' ' + order[0] + (order[1] ? ' ' + order[1].toUpperCase(): '') + ((orders.length - 1) != i ? ',': '');
```
Comment (zerious) [inline lib/Database.js]: Could you please ensure that this isn't open to SQL injection? You could add a "sanitize" method like the "quote" method for order[0], and for order[1], maybe use:
```
var direction = (order[1] || '').replace(/^.*?(asc|desc)?.*?$/, '$1');
```

Comment (zerious) [inline lib/Database.js]: Additionally, there's a mapping between field names and column names. I believe the default is camel case for fields and snake case for columns. Either way, you can look at how the filter column SQL is assembled or SELECT fields and filters, and use something similar for order by fields.

File: test/databaseTest.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  it('empty order should return empty string', function () {
+    var order = [];
+    is.same(query.getOrderBy(order),'');
+  });
+
+  it('undefined order should return empty string', function () {
+    is.same(query.getOrderBy(),'');
+  });
+});
```
Comment (zerious) [inline test/databaseTest.js]: Thanks for adding tests!

File: test/databaseTest.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+describe('Database', function () {
+  Database = Database.extend({
+    init: function() {}
+  });
+  query = new Database();
+  it('should prepare query segment for order when with array content', function () {
+    var order = [
+      ['created'],
+      ['ormy', 'desc'],
+      ['label', 'asc']
```
Comment (zerious) [inline test/databaseTest.js]: Try to order by something like:
```js
var order = [['created;DROP TABLE customers']];
```

Comment (alfredkam) [inline test/databaseTest.js]: added for both string and array validation

string : https://github.com/lighterio/ormy/pull/10/files#diff-024b0d91f5be2c2a6a92ba02162e4ee3R73

array : https://github.com/lighterio/ormy/pull/10/files#diff-024b0d91f5be2c2a6a92ba02162e4ee3R50

**Comment by alfredkam** [comment]
should be gd to merge now

File: lib/Database.js [hunk 4]
```diff
@@ -148,6 +151,45 @@ var Database = module.exports = Class.extend({
...
+  getOrderBy: function (model, orders) {
+    var db = this;
+    if (!orders || (orders && orders.length === 0)) return '';
+
+    if (typeof orders == 'string') {
+      return db.getOrderBy(model, db.breakStringIntoArrays(orders));
+    }
+
+    var query = '';
+    for (var i in orders) {
```
Comment (HasSamE) [inline lib/Database.js]: for..in iteration is slower, and it's also susceptible to Object.prototype changes.

**Pull request merged**